### PR TITLE
feat(GlobalHeader): add upgrade link for plus users

### PR DESCRIPTION
**Pull request by xendke**
### Overview

- Add a link to the app headers when user has a Plus subscription.
- Add a new Icon from [Streamline](https://www.streamlinehq.com/icons/streamline-regular?search=arrow+thick+circle+up)

### PR Checklist

- [x] Related to designs: [Figma](https://www.figma.com/file/WZhdoCkyGqvWvk06bbYSsT/Upgrade-on-pricing-for-Pro-Lite?node-id=0-1&t=w3CMVE6q4brUZYRt-0)
- [x] Related to JIRA ticket: [FLOW-485](https://codecademy.atlassian.net/browse/FLOW-485)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Sign up as a Plus user (you can do this in the Test User Creation Tool by selecting `pro-lite` and also adding a subscription) [Test User Prefill Link](https://pr-35399-monolith.dev-eks.codecademy.com/admin/test_user?prefill_id=64414ffadff6310001c9c228)
2. GlobalHeader should now have an "Upgrade" link that leads to the pricing page
3. Mobile also should have the same link with the new icon

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/35399 'Named link title')  | [Monolith Env](https://pr-35399-monolith.dev-eks.codecademy.com/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/1688 'Named link title')  | [Portal Env](https://tengu.codecademy.com/pricing?PR_ENV=portal-app-pr-1688 'Named link title')   |

[FLOW-485]: https://codecademy.atlassian.net/browse/FLOW-485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.307-alpha.3ce481.0<br/>@codecademy/gamut-labs@35.11.5-alpha.3ce481.0<br/>@codecademy/styleguide@64.2.4-alpha.3ce481.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.307-alpha.ddcd4d.0<br/>@codecademy/gamut-labs@35.11.5-alpha.ddcd4d.0<br/>@codecademy/styleguide@64.2.4-alpha.ddcd4d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://643d68d845d09f2f8c0194ee--gamut-preview.netlify.app>https://643d68d845d09f2f8c0194ee--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/643d68d845d09f2f8c0194ee>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@9.0.3-alpha.17932d.0<br/>@codecademy/gamut-kit@0.6.309-alpha.17932d.0<br/>@codecademy/gamut-labs@35.12.2-alpha.17932d.0<br/>@codecademy/gamut@53.1.2-alpha.17932d.0<br/>@codecademy/monaco@0.7.20-alpha.17932d.0<br/>@codecademy/styleguide@64.3.2-alpha.17932d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://643ec9184aeb2e2bcfa63d9d--gamut-preview.netlify.app>https://643ec9184aeb2e2bcfa63d9d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/643ec9184aeb2e2bcfa63d9d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@9.0.3-alpha.88f1f6.0<br/>@codecademy/gamut-kit@0.6.309-alpha.88f1f6.0<br/>@codecademy/gamut-labs@35.12.2-alpha.88f1f6.0<br/>@codecademy/gamut@53.1.2-alpha.88f1f6.0<br/>@codecademy/monaco@0.7.20-alpha.88f1f6.0<br/>@codecademy/styleguide@64.3.2-alpha.88f1f6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://644032dd76a2fe05e1c625e3--gamut-preview.netlify.app>https://644032dd76a2fe05e1c625e3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/644032dd76a2fe05e1c625e3>Deploy Logs</a></p>

File: packages/gamut-icons/src/svg/regular/arrow-thick-circle-up-icon.svg
Comment (xendke) [inline packages/gamut-icons/src/svg/regular/arrow-thick-circle-up-icon.svg:1]: I wasn't able to get this SVG to resize with the `viewBox` when I downloaded it as "Outline Strokes" 😢

**Review by dreamwasp — APPROVED** [review]
lgtm!

**Pull request merged**